### PR TITLE
Rewrite the dynamic dispatch section to focus on usage rather than implementation.

### PR DESCRIPTION
**Pull request by Ms2ger**

### CODE REVIEW

**Comment by Ms2ger** [comment]
r? @steveklabnik 

```
<njn> Ms2ger: that's definitely an improvement
<njn> Ms2ger: I suggest changing the method name to something other than `method`, because that becomes unclear in the representation section
<njn> Ms2ger: also, is the representation section important to be able to use trait objects? If not, a "you don't really need to know these details" warning would be helpful
<huon> njn: it is helpful to understand certain errors (particularly those around object safety)
<huon> but that's all, AFAIK
```

If you have wording that would be good for that, please let me know.

File: src/doc/trpl/static-and-dynamic-dispatch.md [hunk 1]
```diff
@@ -111,28 +133,42 @@ resulting in less code bloat. However, this comes at the cost of requiring
 slower virtual function calls, and effectively inhibiting any chance of
 inlining and related optimisations from occurring.
 
-Trait objects are both simple and complicated: their core representation and
-layout is quite straight-forward, but there are some curly error messages and
-surprising behaviors to discover.
-
```
Comment (steveklabnik) [inline src/doc/trpl/static-and-dynamic-dispatch.md]: Why remove this?

File: src/doc/trpl/static-and-dynamic-dispatch.md [hunk 2]
```diff
@@ -111,28 +133,42 @@ resulting in less code bloat. However, this comes at the cost of requiring
...
 ### Obtaining a trait object
 
-There's two similar ways to get a trait object value: casts and coercions. If
-`T` is a type that implements a trait `Foo` (e.g. `u8` for the `Foo` above),
-then the two ways to get a `Foo` trait object out of a pointer to `T` look
-like:
+As we saw above, it is possible to create a trait object by casting the
+reference explicitly:
```
Comment (steveklabnik) [inline src/doc/trpl/static-and-dynamic-dispatch.md]: I'm not totally sure that I like using this concept before introducing it.

File: src/doc/trpl/static-and-dynamic-dispatch.md [hunk 3]
```diff
@@ -143,8 +179,33 @@ This operation can be seen as "erasing" the compiler's knowledge about the
...
+
 ### Representation
 
+The methods of the trait can be called on a trait object via a special record
+of function pointers traditionally called a 'vtable' (created and managed by
+the compiler).
+
+Trait objects are both simple and complicated: their core representation and
+layout is quite straight-forward, but there are some curly error messages and
+surprising behaviors to discover.
```
Comment (steveklabnik) [inline src/doc/trpl/static-and-dynamic-dispatch.md]: oh, this moved here, cool :+1:

**Comment by steveklabnik** [comment]
I like this overall, my only beef is using that syntax before explaining it. What do you think?

**Comment by steveklabnik** [comment]
r=me after a squash

**Pull request merged**